### PR TITLE
support spring boot3

### LINKED ISSUE #81: Spring boot 3.3.3 使用时报错
- [ ] 我已经检查过[discussions](https://github.com/apolloconfig/apollo-java/discussions)
- [ ] 我已经搜索过[issues](https://github.com/apolloconfig/apollo-java/issues)
- [ ] 我已经仔细检查过[FAQ](https://www.apolloconfig.com/#/zh/faq/common-issues-in-deployment-and-development-phase)

**描述bug**

在 spring boot 3.3.3 中使用 apollo-client 时，报如下错误：

o.s.c.a.ConfigurationClassPostProcessor  : Cannot enhance @Configuration bean definition 'com.ctrip.framework.apollo.spring.boot.ApolloAutoConfiguration' since its singleton instance has been created too early. The typical cause is a non-static @Bean method with a BeanDefinitionRegistryPostProcessor return type: Consider declaring such methods as 'static' and/or marking the containing configuration class as 'proxyBeanMethods=false'.

**复现**

通过如下步骤可以复现：

1. 
2. 
3. 
4. 

**期望**

简介明了地描述你希望正常情况下应该发生什么

**截图**

如果可以，附上截图来描述你的问题

### 额外的细节和日志

- 版本：
- 错误日志
- 配置：
- 平台和操作系统

### PR DESCRIPTION
**Pull request by qiujiayu**
## What's the purpose of this PR

XXXXX

## Which issue(s) this PR fixes:
Fixes #81 

## Brief changelog

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Read the [Contributing Guide](https://github.com/apolloconfig/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit tests to verify the code.
- [x] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [x] Update the [`CHANGES` log](https://github.com/apolloconfig/apollo-java/blob/master/CHANGES.md).

## Summary by CodeRabbit

- **New Features**
	- Added support for Spring Boot 3, enhancing compatibility with the latest framework version.
	- Improved auto-configuration capabilities for better integration with the Spring framework.
  
- **Bug Fixes**
	- Resolved the "Cannot enhance @Configuration bean definition" issue, improving functionality.

### CODE REVIEW

**Comment by qiujiayu** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by nobodyiam — COMMENTED** [review]
Thanks. Would you please also help to update the CHANGES.md?

**Comment by qiujiayu** [comment]
> Thanks. Would you please also help to update the CHANGES.md?

done

File: CHANGES.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Release Notes.
 Apollo Java 2.4.0
 
 ------------------
-* 
+* [support spring boot3](https://github.com/apolloconfig/apollo-java/pull/82)
```
Comment (nobodyiam) [inline CHANGES.md:8]:
```suggestion
* [Fix the Cannot enhance @Configuration bean definition issue](https://github.com/apolloconfig/apollo-java/pull/82)
```

**Comment by nobodyiam** [comment]
As this is a warning message and occurs not only in spring boot 3 but also spring boot 2, I changed the description to `Fix the Cannot enhance @Configuration bean definition issue`.

**Review by nobodyiam — APPROVED** [review]
LGTM

**Pull request merged**